main.tf

```diff
@@ -7,11 +7,43 @@ terraform {
   }
 }
 
+# Configure the AWS Provider
+provider "aws" {
+  region = "us-east-1"
+}
+
 resource "random_string" "deployment_id" {
   count = "${var.deployment_id != "" ? 0 : 1}"
 
-  length   = 8
+  length = 8
+  special  = false
+  upper = false
+  numeric = false
+}
+
+resource "random_string" "instance_name_suffix" {
+  length = 8
   special  = false
   upper = false
   numeric = false
 }
+
+data "aws_ami" "latest_al2_linux_ami" {
+  most_recent = true
+  owners = ["amazon"]
+
+  filter {
+    name = "name"
+    values = ["amzn2-ami-hvm-*-x86_64-ebs"]
+  }
+}
+
+module "controller_vm" {
```

Comment:
note - module controller_vm here is just an example to show how template could be used